.codebuild/build.sh

```diff
@@ -9,94 +9,219 @@ send_slack_message() {
+        for var in "${!mainnet_@}"; do
+            export ${var#mainnet_}="${!var}"
+        done
+        make deploy-lambdas-mainnet;
+        send_slack_message "Rollback performed on mainnet-production to: ${GIT_REF_TO_DEPLOY}";
+        exit=true;
+    fi;
+
+    # Checks whether there is a file called "rollback_testnet_production", which is used by our other CodeBuild to indicate that this is a testnet-production rollback
+    if [ -f "rollback_testnet_production" ]; then
+        # Gets all env vars with `testnet_` prefix and re-exports them without the prefix
+        for var in "${!testnet_@}"; do
+            export ${var#testnet_}="${!var}"
+        done
+        make deploy-lambdas-testnet;
+        send_slack_message "Rollback performed on testnet-production to: ${GIT_REF_TO_DEPLOY}";
+        exit=true;
+    fi;
+
+    if [ "$exit" = true ]; then
+    echo "Rollbacks performed successfully. Exiting now.";
+    exit 0;
+    fi
+
+    if expr "${GIT_REF_TO_DEPLOY}" : "master" >/dev/null; then
+        # Gets all env vars with `dev_` prefix and re-exports them without the prefix
+        for var in "${!dev_@}"; do
+            export ${var#dev_}="${!var}"
+        done
+
+        make migrate;
+        make build-daemon;
+        make deploy-lambdas-dev-testnet;
+        # The idea here is that if the lambdas deploy fail, the built image won't be pushed:
+        make push-daemon;
+
+    elif expr "${GIT_REF_TO_DEPLOY}" : "v[0-9]\+\.[0-9]\+\.[0-9]\+-rc\.[0-9]\+" >/dev/null; then
+        # Gets all env vars with `mainnet_staging_` prefix and re-exports them without the prefix
+        for var in "${!mainnet_staging_@}"; do
+            export ${var#mainnet_staging_}="${!var}"
+        done
+
+        echo $GIT_REF_TO_DEPLOY > /tmp/docker_image_tag
+        make migrate;
+        make build-daemon;
+        make deploy-lambdas-mainnet-staging;
+        # The idea here is that if the lambdas deploy fail, the built image won't be pushed:
+        make push-daemon;
+        send_slack_message "New version deployed to mainnet-staging: ${GIT_REF_TO_DEPLOY}"
+    elif expr "${GIT_REF_TO_DEPLOY}" : "v.*" >/dev/null; then
+        echo $GIT_REF_TO_DEPLOY > /tmp/docker_image_tag
+        make build-daemon;
+
+        # --- Testnet ---
+        # Gets all env vars with `testnet_` prefix and re-exports them without the prefix
+        for var in "${!testnet_@}"; do
+            export ${var#testnet_}="${!var}"
+        done
+
+        make migrate;
+        make deploy-lambdas-testnet;
+
+        # Unsets all the testnet env vars so we make sure they don't leak to other deploys
+        for var in "${!testnet_@}"; do
+            unset ${var#testnet_}
+        done
+
+        # --- Mainnet ---
+        # Gets all env vars with `mainnet_` prefix and re-exports them without the prefix
+        for var in "${!mainnet_@}"; do
+            export ${var#mainnet_}="${!var}"
+        done
+        make migrate;
+        make deploy-lambdas-mainnet;
```

Comment:
Why did you remove `make build-daemon;`? Do you need to run only once for both envs?

Reply:
Yes, both Testnet and Mainnet use the same Docker images currently. Pushing them twice as we did before as unnecessary.